lib/anidb/media_category_assoc_importer.rb

```diff
@@ -0,0 +1,91 @@
+class AnidbAssocMediaCategoryImport
+  class ImportFile
+    attr_reader :data
+
+    def initialize(base_media_url, anime_assoc_file, genre_map_file)
+      # Loading the JSON
+      utc = Time.now.to_i.to_s
+      anime_json_file = open(
+        base_media_url +
+        anime_assoc_file +
+        '?' +
+        utc
+      ).read
+      @data = JSON.parse(anime_json_file)
+      genre_categeory_map = open(
+        base_media_url +
+        genre_map_file +
+        '?' +
+        utc
+      ).read
+      gc_data = JSON.parse(genre_categeory_map)
+      gc_data.update(gc_data) { |_, v| Category.where(title: v)[0] if v }
+      @genre_category_model = gc_data
+    end
+
+    def associate_media_categores(media)
+      genres = media.genres.map(&:name)
+      return unless genres.empty?
+      categories = genres.map { |g|
+        @genre_category_model[g]
+      }.compact
+      if categories
+        media.categories = categories
+        media.save
+      end
+    end
+
+    def associate_manga!
+      puts 'Associating Manga Categories From Kitsu Genre Map'
+      manga = Manga.includes(:genres).all
+      manga.each do |m|
+        associate_media_categores(m)
+      end
+    end
+
+    def associate_empty_anime(exclude_ids)
+      puts 'Associating Excluded Anime With Categories From Kitsu Genre Map'
+      anime = Anime.includes(:genres).where.not(id: exclude_ids)
+      anime.each do |a|
+        associate_media_categores(a)
+      end
+    end
+
+    def associate_initial_anime!
+      exclude_ids = []
+      puts 'Associating Anime With Categories From AniDB Dump'
+      data.each do |unfiltered_anime|
+        unfiltered_anime = unfiltered_anime.deep_symbolize_keys
+        next unless unfiltered_anime[:tags]
+        mapping_object = {
+          title: unfiltered_anime[:canonical],
+          episode_count: unfiltered_anime[:episode_count]
+        }
+        puts 'looking up -> ' + unfiltered_anime[:canonical]
```

Comment:
I think there's a way to get the MAL ID from the AniDB dump, which might give a better result.  I'm not gonna lie and say `Mapping.guess` doesn't suck, because it massively sucks.